manager/src/main/java/io/gingersnapproject/hotrod/CommandProcessor.java

```diff
@@ -0,0 +1,303 @@
+   }
+
+   public void put(GingersnapHeader header, byte[] key, byte[] value) {
+      if (header.hasFlag(ProtocolFlag.ForceReturnPreviousValue)) {
+         writeException(header, new UnsupportedOperationException("previous value is not supported!"));
+         return;
+      }
+      var cache = getOrCreateMap(header.getCacheName());
+
+      try {
+         cache.put(new WrappedByteArray(key), new WrappedByteArray(value));
+         writeSuccess(header);
+      } catch (Throwable t) {
+         writeException(header, t);
+      }
+   }
+
+   public void get(GingersnapHeader header, byte[] key) {
+      var cache = getOrCreateMap(header.getCacheName());
+
+      try {
+         WrappedByteArray wba = cache.getIfPresent(new WrappedByteArray(key));
+         if (wba == null) {
+            writeNotExist(header);
+         } else {
+            writeResponse(header.encoder().valueResponse(header, null, channel, OperationStatus.Success, wba.getBytes()));
+         }
+      } catch (Throwable t) {
+         writeException(header, t);
+      }
+   }
+
+   public void putIfAbsent(GingersnapHeader header, byte[] key, byte[] value) {
+      if (header.hasFlag(ProtocolFlag.ForceReturnPreviousValue)) {
+         writeException(header, new UnsupportedOperationException("previous value is not supported!"));
+         return;
+      }
+      var cache = getOrCreateMap(header.getCacheName());
+
+      try {
+         WrappedByteArray wba = cache.asMap().putIfAbsent(new WrappedByteArray(key), new WrappedByteArray(value));
+         if (wba == null) {
+            this.writeSuccess(header);
+         } else {
+            writeResponse(header.encoder().emptyResponse(header, null, channel, OperationStatus.OperationNotExecuted));
+         }
+      } catch (Throwable t) {
+         writeException(header, t);
+      }
+   }
+
+   public void remove(GingersnapHeader header, byte[] key) {
+      if (header.hasFlag(ProtocolFlag.ForceReturnPreviousValue)) {
+         writeException(header, new UnsupportedOperationException("previous value is not supported!"));
+         return;
+      }
+      var cache = getOrCreateMap(header.getCacheName());
+
+      try {
+         var prev = cache.asMap().remove(new WrappedByteArray(key));
+         if (prev == null) {
+            this.writeNotExist(header);
+         } else {
+            this.writeSuccess(header);
+         }
+      } catch (Throwable t) {
+         writeException(header, t);
+      }
+   }
+
+   public void ping(GingersnapHeader header) {
+      this.writeResponse(header.encoder().pingResponse(header, null, this.channel, OperationStatus.Success));
+   }
+
+   public void size(GingersnapHeader header) {
+      var cache = maps.getMaps().get(header.getCacheName());
+      var size = cache == null ? -1 : cache.estimatedSize();
+      this.writeResponse(header.encoder().unsignedLongResponse(header, null, this.channel, size));
```

Comment:
Oops, this doesn't work for the -1. To be honest we will want to throw an error instead when the map isn't present. One of the features missing is we probably shouldn't define maps on a per access but have them be configured by config map solely.